realm/realm-library/src/main/java/io/realm/RealmBaseRecyclerAdapter.java

```diff
@@ -0,0 +1,195 @@
+
+    protected LayoutInflater inflater;
+    protected Context context;
+    private OrderedRealmCollection<T> adapterData;
+    private RealmChangeListener listener;
+
+    private boolean hasAutoUpdates;
+
+    public RealmBaseRecyclerAdapter(Context context, OrderedRealmCollection<T> data, boolean autoUpdate) {
+        if (context == null) {
+            throw new IllegalArgumentException("Context cannot be null");
+        }
+
+        this.context = context;
+        this.adapterData = data;
+        this.inflater = LayoutInflater.from(context);
+        this.hasAutoUpdates = autoUpdate;
+        if (autoUpdate) {
+           listener = new RealmChangeListener() {
+               @Override
+               public void onChange() {
+                   notifyDataSetChanged();
+               }
+           };
+        }
+    }
+
+    @Override
+    public void onAttachedToRecyclerView(final RecyclerView recyclerView) {
+        super.onAttachedToRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            addListener(adapterData);
+        }
+    }
+
+    @Override
+    public void onDetachedFromRecyclerView(final RecyclerView recyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            removeListener(adapterData);
+        }
+    }
+
+    /**
+     * Returns the current ID for an item. Note that item IDs are not stable so you cannot rely on the item ID being the
+     * same after {@link #notifyDataSetChanged()} or {@link #updateData(OrderedRealmCollection)} has been called.
+     *
+     * @param index position of item in the adapter.
+     * @return current item ID.
+     */
+    @Override
+    public long getItemId(final int index) {
+        return index;
+    }
+
+    @Override
+    public int getItemCount() {
+        return isDataValid() ? adapterData.size() : 0;
+    }
+
+    /**
+     * Returns the item associated with the specified position.
+     *
+     * @param index index of the item.
+     * @return the item at the specified position.
+     */
+    public T getItem(int index) {
+        return isDataValid() ? adapterData.get(index) : null;
+    }
+
+    /**
+     * Removes the item associated with the specified position. This also deletes the objects from the underlying Realm.
+     *
+     * @param index index of the item to remove.
+     */
+    public void removeItem(int index) {
+        if (isDataValid()) {
+            getRealm().beginTransaction();
+            adapterData.deleteFromRealm(index);
+            getRealm().commitTransaction();
+
+            // There is no need to call notify methods if we have listener
+            if (!hasAutoUpdates) {
+                notifyItemRemoved(index);
+            }
+        }
+    }
+
+    /**
+     * Clears all data from the adapter. This also deletes the objects from the underlying Realm.
+     */
+    public void clear() {
```

Comment:
This method is not part of RecyclerView.Adapter interface. Is there a good reason for adding it? The default way of doing this would be to clear the backing collection and then call `notifyDataSetChanged`.